aidb/engine/approx_aggregate_engine.py

```diff
@@ -54,9 +53,22 @@ async def execute_aggregate_query(self, query: Query):
             error target. Try running without the error target if you are certain you want to run this query.'''
         )
 
-      agg_type = query.get_agg_type
-      estimator = self._get_estimator(agg_type)
-      num_samples = self.get_additional_required_num_samples(query, sample_results, estimator)
+      num_samples = []
+      agg_index = 0
+      # loop to determine max number of samples required
+      for agg_type, columns_per_agg in query.aggregated_columns_and_types:
+        estimator = self._get_estimator(agg_type)
+        for _ in columns_per_agg:
+          num_samples.append(
+            self.get_additional_required_num_samples(

```

Comment:
This doesn't seem correct to me, since if we want the confidence interval to apply to each aggregation individually, we need to adjust the confidence

Reply:
I've used Bonferroni correction and changed code